src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```
@@ -0,0 +1,109 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static RegistrationService registrationService;
+
+    private User createUser(String login, String password, Integer age) {
+        User newUser = new User();
+        newUser.setLogin(login);
+        newUser.setPassword(password);
+        newUser.setAge(age);
+        return newUser;
+    }
+
+    @BeforeAll
+    static void setUp() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @Test
+    void register_nullUser_notOk() {
+        assertThrows(RuntimeException.class, () -> {
+            User registeredUser = registrationService.register(null);;
+        });
+    }
+
+    @Test
+    void register_nullLoginUser_notOk() {
+        User user = createUser(null, "password", 21);
+        assertThrows(RuntimeException.class, () -> {
+            User registeredUser = registrationService.register(user);;
+        });
+    }
+
+    @Test
+    void register_nullPasswordUser_notOk() {
+        User user = createUser("login", null, 21);
+        assertThrows(RuntimeException.class, () -> {
+            User registeredUser = registrationService.register(user);;
+        });
+    }
+
+    @Test
+    void register_nullAgeUser_notOk() {
+        User user = createUser("login", "password", null);
+        assertThrows(RuntimeException.class, () -> {
+            User registeredUser = registrationService.register(user);;
+        });
+    }
+
+    @Test
+    void register_existingLoginUser_notOk() {
```

should throw an exception here